src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -303,6 +304,37 @@ static bool can_reshape_be_optimized(const reshape_node& node) {
     return node.is_in_place() && node.get_fused_activations_funcs().empty();
 }
 
+#ifdef ENABLE_ONEDNN_FOR_GPU
+// pattern #1: input -> (bfyx) -> permute -> (byxf) -> oneDNN convolution
+//      - permute node reorders is [0, 3, 1, 2] (bfyx -> byxf).
+//      - input layout of onednn convolution is byxf.
+// pattern #2: oneDNN convolution convolution -> (byxf) -> permute -> (bfyx) -> output
+//      - permute node reorders is [0, 2, 3, 1] (byxf -> bfyx).
+//      - output layout of permute is byxf
+static bool can_permute_be_optimized(const permute_node& node) {
+    if (node.is_output() || node.get_dependencies().size() != 1 || node.get_users().size() != 1
+        || !node.get_output_layout().is_static()
+        || !node.get_dependency(0).get_output_layout().is_static())
+        return false;
+
+    if (node.get_output_layout().format != cldnn::format::byxf
+        && node.get_output_layout().format != cldnn::format::bfyx)
+        return false;
+
+    // permute - convolution
+    auto next = node.get_users().front();
+    if (next->is_type<convolution>() && next->get_preferred_impl_type() == impl_types::onednn)
+        return node.is_reverse_rotating_except_batch();
+
+    // convolution - permute
+    auto& prev = node.get_dependency(0);
+    if (prev.is_type<convolution>() && prev.get_preferred_impl_type() == impl_types::onednn)
+        return node.is_rotating_except_batch();
+
+    return false;
+}
```

Comment:
This is checking permute layer only. Isn't it breaking the integrity of graph if we don't check reorder?

For example, From the pattern below, if we don't have reorder, permute should not be optimized out, right? If so, you need to confirm that reorder is coming after permute too..

|  123 | conv2d_109/Conv2D                     | Convolution | 163          | [122]            | [32, 1024, 13, 13]  | bs_fs_yx_b32_f32 | I8       | [32, 255, 13, 13]   | bs_fs_yx_b32_f16 | FP16     | jit:ir__i8                       |
|  124 | conv2d_109/BiasAdd/Add                | Permute     | 136          | [123]            | [32, 255, 13, 13]   | bs_fs_yx_b32_f16 | FP16     | [32, 13, 13, 255]   | bs_fs_yx_b32_f16 | FP16     | permute_ref__f16                 |
|  125 | conv2d_109/BiasAdd:0                  | Reorder     | 747          | [124]            | [32, 13, 13, 255]   | bs_fs_yx_b32_f16 | FP16     | [32, 13, 13, 255]   | bfyx             | FP32     | reorder_data__f32                |

Reply:
This code has been removed.